hy/compiler.py

```diff
@@ -2312,6 +2316,19 @@ def compile_function_def(self, expression):
 
         return ret
 
+    @builds("return")
+    @checkargs(max=1)
+    def compile_return(self, expr):
+        value = None
+        ret = Result()
+        if len(expr) > 1:
+            value = self.compile(expr[1])
```

Comment:
I just realized that I can get rid of one of the assignments, because `ret.force_expr` works, so I did that.